### PR TITLE
Method for addon to detect external player

### PR DESCRIPTION
**Pull request by angelblue05**
## Description

Add a way for add-ons to detect if an external player is used for playback. 

## Motivation and Context

This allows for a more precise detection, especially when checking the current playback position. When an external player is used, the position remains 0.0 until the file is considered watched, which then returns 1.0. It is currently impossible to tell the difference between a normal position tick and the external player position tick. This is why I made this new addition.

## How Has This Been Tested?

I compiled Kodi windows x64 to test.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/interfaces/legacy/Player.h [hunk 1]
```diff
@@ -449,6 +449,21 @@ namespace XBMCAddon
...
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_Player
+      /// @brief \python_func{ isExternalPlayer() }
+      ///-----------------------------------------------------------------------
+      /// Check for external player.
+      ///
+      /// @return                    True if kodi is playing using an
+      ///                            external player.
+      ///
```
Comment (ronie) [inline xbmc/interfaces/legacy/Player.h]: could you please add a note like this one:
https://github.com/xbmc/xbmc/blob/e53988394230379d0e00ab1812fc3920fdaa6723/xbmc/interfaces/legacy/Control.h#L1106

so we can keep track on in which kodi version new functions were introduced.

**Comment by ronie** [comment]
looks good to me.

File: xbmc/interfaces/legacy/Player.h (hunk 1 above)
Comment (angelblue05) [reply xbmc/interfaces/legacy/Player.h]: Sure thing!

**Comment by angelblue05** [comment]
I don't think Jenkins likes it when I update a commit...

**Comment by angelblue05** [comment]
@ronie 
I know my commit works. Jenkins fails the check because I updated my original commit (sometimes it works). How do I force it to do another check, or is it a big deal? Let me know. Thank you!

**Comment by ronie** [comment]
looks like windows builds are just broken. not related to your PR.

**Comment by Rechi** [comment]
@ronie good to get merged?

**Comment by ronie** [comment]
yup, thx!

**Pull request merged**